docs/debug.md

```diff
@@ -0,0 +1,192 @@
+    * **background_span_fetch_failure_count** - number of errors of span fetch by background fetcher
+    * **background_span_fetch_count** - number of spans fetched by background fetcher
+    * **background_fetch_work_queue_size** - number of items in the work queue of background fetcher
+    * **operation_duration_background_fetch** - time in milliseconds to complete background fetch for a layer
+    * Individual FUSE operation failure counts
+      * fuse_node_getattr_failure_count
+      * fuse_node_listxattr_failure_count
+      * fuse_node_lookup_failure_count
+      * fuse_node_open_failure_count
+      * fuse_node_readdir_failure_count
+      * fuse_file_read_failure_count
+      * fuse_file_getattr_failure_count
+      * fuse_whiteout_getattr_failure_count
+      * fuse_unknown_operation_failure_count
+
+# Common Scenarios
+
+Although creating an exhaustive list of all possible errors that may occur when using `SOCI` would not be feasible, we will attempt to outline and provide some guidance to address common scenarios that may occur both during `rpull` and throughout the lifetime of a running container. 
+
+> _Note: To allow for more verbose logging you can set the `--log-level` flag to `debug` when starting the snapshotter._
+
+## Rpull 
+
+During `rpull` the image manifest, config, and layers without zTOCs' are fetched from the remote registry. Layers that have a zTOC are mounted as a `FUSE` file system
+
+Below are a list of common error paths that may occur in this phase:
+
+### No lazy-loading
+
+If you notice that all layers are being fetched for an image or that fuse mounts are not being created for layers with a zTOC than that means that remote mounting has failed for those layers.
+
+Once you inspect the logs you should come across an error log that contains the message `failed to prepare remote snapshot` with an `error` key describing the error that was propagated up within the snapshotter.
+
+Some possible error keys include:
+
+* `skipping mounting layer <layer_digest> as FUSE mount: no zTOC for layer`
+  
+  This "error" message is not really indicative of a true error, but rather implies that the current layer does not have an associated zTOC.
+  This is expected for layers that do not meet the minimum-layer size criteria established when creating the soci-index/zTOCs'. 
+
+* `unable to fetch SOCI artifacts: <error>`
+  
+  This error indicates that the soci index along with the corresponding zTOC could not be fetched from the remote registry. This can be for a variety of different reasons. The most common reason is that the resolver could not authenticate against the remote registry. The snapshotter uses the docker resolver to resolve blobs in the remote so you must authenticate with docker first. If you are using `ECR` as your registry you can:
+  ```
+  export ECR_PASS=$(aws ecr get-login-password --region <region>)
+  echo $ECR_PASS | sudo docker login -u AWS --password-stdin $ECR_REGISTRY 
+  ```
```

Comment:
an empty line before and after code block, and add language annotation (e.g., shell)